### ISSUE TITLE
Can adjoint calculations be run with MPI build of meep?

### ISSUE THREAD

**Issue by DurhamSmith** [NanoComp/meep#1863]
When installing the MPI build of meep and trying to run `import adjoint as mpa` I get `ModuleNotFoundError: No module named 'adjoint'
`. Is this because meep does not support the adjoint method for MPI builds or is it because the installation is faulty. 

Additionally it seems that the adjoint method optimization tests are failing for me with the errors (on my non-MPI build):

```
======================================================================
ERROR: test_damping (__main__.TestAdjointSolver)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/user/software/meep/python/tests/test_adjoint_solver.py", line 492, in test_damping
    adjsol_obj, adjsol_grad = adjoint_solver_damping(p, frequencies)
  File "/home/user/software/meep/python/tests/test_adjoint_solver.py", line 303, in adjoint_solver_damping
    matgrid = mp.MaterialGrid(mp.Vector3(Nx,Ny),
TypeError: __init__() got an unexpected keyword argument 'damping'

======================================================================
FAIL: test_offdiagonal (__main__.TestAdjointSolver)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/user/software/meep/python/tests/test_adjoint_solver.py", line 538, in test_offdiagonal
    self.assertClose(adj_scale,fd_grad,epsilon=tol)
  File "/home/user/software/meep/python/tests/utils.py", line 28, in assertClose
    self.assertLessEqual(diff_norm, epsilon * np.maximum(x_norm, y_norm), msg)
AssertionError: 8.687616300904509e-05 not less than or equal to 9.137056711620364e-06 : 

----------------------------------------------------------------------
Ran 6 tests in 309.977s

FAILED (failures=1, errors=1)

```
Is this a failure in the current code base or should I diagnose my installation?

**Comment by smartalecH** [comment]
Sounds like your installation is off. 

The adjoint code works with MPI (as seen with the current CI which builds and runs the adjoint tests with MPI)

**Comment by ahoenselaar** [comment]
Shouldn't that be `import meep.adjoint as mpa`?

**Comment by stevengj** [comment]
As explained above, this is a problem in your script/installation and not in Meep.

**Comment by DurhamSmith** [comment]
Thanks. Here is some information for anyone that experiences similar issues.

1) `import meep.adjoint as mpa`
I was running `test_adjoint_solver.py' which tries to:
```
try:
    import meep.adjoint as mpa
except:
    import adjoint as mpa
```
` import meep.adjoint as mpa` was failing since I did not have the required libraries. If you are using pymeep provided by anaconda hese can be installed with 
```
conda install -c conda-forge scipy
conda install -c conda-forge autograd
conda install -c conda-forge matplotlib
```

Note that conda-forge is required otherwise you will run into issues with numpy using MKL not OpenBLAS (check https://meep.readthedocs.io/en/latest/Installation/).

2) `TypeError: __init__() got an unexpected keyword argument 'damping'`

I encountered this because I just cloned the repo and ran the tests from there  as condas provided  meep did not pull in any tests (or atleast `find ~/ -name test_adjoint_solver.py` returned nothing. The problem here is the git repo is ahead of the conda package. To get around this just checkout the commit that corresponds to the conda package and your tests should run fine.
The other way would be building from source, which is what I plan to do in the near future.
